### PR TITLE
Mobile: improved photo taking experience

### PR DESCRIPTION
**Pull request by shinglyu**
1. Fixed a bug in the camera that allows the camera button to be pressed repeatedly, causing multiple photos to be taking in a short period of time.
2. Add an alt text for the newly taken photo `Photo taken at <localized date time>`. This gives a more meaningful title for a new note. In the past the alt text is empty so the title will become the note ID (see attachment).

## Before: 
before

## After

https://user-images.githubusercontent.com/3250983/156072655-aa91f4e1-6008-40d6-a741-b25c590303a0.mp4

(The camera is blacked out for privacy)

### CODE REVIEW

File: packages/app-mobile/components/screens/Note.tsx [hunk 1]
```diff
@@ -738,6 +738,7 @@ class NoteScreenComponent extends BaseScreenComponent {
 				didCancel: false,
 				error: null,
 				type: 'image/jpg',
+				name: `Photo taken at ${time.formatMsToLocal(Date.now())}`,
```
Comment (CalebJohn) [inline packages/app-mobile/components/screens/Note.tsx:741]:
```suggestion
				name: `${_('Photo taken at')} ${time.formatMsToLocal(Date.now())}`,
```

User facing strings need to be localized. This is done using the `_` function which is already imported at the top of this file.

**Review by CalebJohn — CHANGES_REQUESTED** [review]

File: packages/app-mobile/components/screens/Note.tsx (hunk 1 above)
Comment (shinglyu) [reply packages/app-mobile/components/screens/Note.tsx]: Done! Dumb question: I refreshed the translation using `yarn buildTranslations`, how do I also update the translation for mobile?

Comment (laurent22) [reply packages/app-mobile/components/screens/Note.tsx]: Thanks, but actually we don't update the translations in PR as that creates too many changes, so please could you revert this? We'll update the translations in a separate commit later on.

Comment (shinglyu) [reply packages/app-mobile/components/screens/Note.tsx]: Reverted! Thank you.

**Comment by laurent22** [comment]
Thanks for the update but come to think of it, I wonder if we want this extra alt text? The note timestamp and even the resource timestamp would already give that information, so it seems redundant. Also alt text should describe the photo I believe while here it's just metadata of little value.

I agree in general that having some text here would be nice, but I don't know what it could be, and I'm not convinced a timestamp is a good idea.

**Comment by shinglyu** [comment]
I can simplify it to just "Photo" or something similar. The main purpose for adding the alt is to avoid the title being set to the note ID automatically. If I see a note title "Photo" I know I need to go in and set a correct title. But if it's a random ID it's really confusing (and ugly). 

If you think the alt text bit needs more discussion. I can revert that and just leave the fix for the camera button. That one is a clear bug and affects photo taking experience.

**Comment by laurent22** [comment]
Hmm, I see what you mean and I don't have this problem, because when I create a note with just one photo I always manually set the title, since obviously the app cannot guess.

I agree the title set to a guid is ugly though, but then maybe we should just fix the function that sets the title and makes it ignore these IDs.

**Comment by shinglyu** [comment]
OK. Let me split the title part into a new PR? That change will affect desktop and cli as well, so it's better to start a new thread.

**Comment by laurent22** [comment]
Thanks for the fix @shinglyu. And for the title issue, yes a second PR would be better. I guess it could be something that filters out these IDs specifically, so that we don't get too many false positives.

**Pull request merged**